### PR TITLE
fix: nftshapes error retry

### PR DESCRIPTION
**Pull request by AjimenezDCL**
#1968

The OpenSea API requests allow us to fetch the info of multiple NFTs in a single request. The downside is that if any of the NFTs is invalid the whole request wont return anything (not even an error). So when dealing with that situation, we split the failed request into 2 smaller requests to narrow down the ofender NFT.

### CODE REVIEW

**Review by D4rWiNSS — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 1]
```diff
@@ -87,19 +87,17 @@ void Awake()
         InitializePerlinNoise();
     }
 
-    void Update()
-    {
-        hqTextureHandler?.Update();
-    }
+    void Update() { hqTextureHandler?.Update(); }
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs:90]: I have seen this "one line function" style in several PRs... is this being applied by the new automated code-style pre-commits? I guess it is something very personal, but I'm not a big fan of this kind of style, even if the function has only 1 line.

File: unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs [hunk 2]
```diff
@@ -127,28 +143,39 @@ IEnumerator Fetch(float delayRequest)
...
-                    if (OpenSeaRequestController.VERBOSE) Debug.Log($"RequestGroup: Request to OpenSea {url}");
+                    if (OpenSeaRequestController.VERBOSE)
+                        Debug.Log($"RequestGroup: Request to OpenSea {url}");
                     yield return request.SendWebRequest();
 
                     AssetsResponse response = null;
 
                     if (!request.isNetworkError && !request.isHttpError)
                     {
+                        Debug.Log(request.downloadHandler.text);
```
Comment (sandrade-dcl) [inline unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs]: Shouldn't we put this log inside a VERBOSE conditional?

**Review by sandrade-dcl — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs (hunk 1 above)
Comment (AjimenezDCL) [reply unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs:90]: Yeah, is the automated code-style. I'm not a big fan of it either but after several tries to fix it without messing anything else up, I gave up hahaha. 

I can give it another try in the near-future.

File: unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs (hunk 2 above)
Comment (AjimenezDCL) [reply unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs]: This shouldnt have been commited hahaha, nice catch!

File: unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs [hunk 3]
```diff
@@ -91,6 +95,18 @@ public OpenSeaRequestGroup(float delayRequest, Action<OpenSeaRequestGroup> onGro
             fetchRoutine = CoroutineStarter.Start(Fetch(delayRequest));
         }
 
+        public OpenSeaRequestGroup(float delayRequest, IEnumerable<KeyValuePair<string, OpenSeaRequest>> existentRequests, Func<float> getRetryDelay)
```
Comment (pbosio) [inline unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs]: If we are using this constructor only inside `OpenSeaRequestGroup` class, maybe we can make it private?

**Review by pbosio — APPROVED** [review]
Great work! I didn't though it was possible to fix this issue without recoding all of these classes  :P
Just a minor, non-important suggestion

**Pull request merged**